change working directory

antondollmaier opened the pull request:
fixes #2508

When running as cron, pg_dump might complain about permissions in the current working directory.

changing directory to the BACKUP_SCRIPT_PATH solves this.

<!--
Hi there - a lot of love for starting a Pull Request 😍. Please ensure the following things before creation - thank you!

- Create your Pull Request against the develop branch. We don't accept changes to stable branches.
- Add a reference to the issue you are trying to fix. Just add a # and the number.
- Don't run `bundle update`. It's a honorable thought of you but some dependency versions (e.g. pg) are broken 😢 Just use `bundle install` if you introduce new dependencies instead.
- Run `rubocop` and `coffeelint` on your changes to respect the code style guide.
- Add tests for your changes. Feel free to ask for support. We are happy to help you.

* The upper textblock will be removed automatically when you submit your Pull Request *
-->

Commit:
change working directory

When running as cron, pg_dump might complain about permissions in the current working directory.

changing directory to the BACKUP_SCRIPT_PATH solves this.

MrGeneration commented:
Thanks for your contribution!

As already mentioned in #2508 I personally don't see this as a real problem, as you should rather work with the crontab of the Zammad-User instead of root.

I will therefor not merge this pull request, as I don't see the benefit (no offense!).

thorsteneckel commented:
@monotek I just saw that you reopened this PR. So you think we should merge it? What do we need to do next?

monotek commented:
https://github.com/zammad/zammad/issues/2508#issuecomment-470527571

MrGeneration commented on contrib/backup/zammad_backup.sh:
```diff
@@ -6,6 +6,9 @@
 BACKUP_SCRIPT_PATH="$(dirname $(realpath $0))"
 
+# chdir to run as Cron
+cd $BACKUP_SCRIPT_PATH;
```
I'd prefer `cd "${BACKUP_SCRIPT_PATH}"` at this point.
Other than that, it's fine to me.

Verified it from my end.

martini commented:
closed by accident - re-open

MrGeneration committed:
optimized the place

Optimized when we run the change dir, because we only need it if everything is good to go.

I'll now add a remark internally so that I won't to it a fourth time.

MrGeneration commented:
@thorsteneckel this is ready for merging. :-)

MrGeneration commented:
We currently can't merge PRs from Github, this is why I'm closing this PR for the sake of our internal MR no. 866 

Thank you very much for your contribution!
